### PR TITLE
npkill: init at 0.12.2

### LINKED ISSUE #326343: Package request: npkill
**Project description**

A tool that lets you find all the `node_modules` directories on your system and see how much space they use. You can then choose which ones to delete to free up space.

**Metadata**

* homepage URL: https://github.com/voidcosmos/npkill
* source URL: https://github.com/voidcosmos/npkill
* license: mit
* platforms: linux, darwin, windows

### PR DESCRIPTION
**Pull request by AveryanAlex**
## Init npkill at 0.12.2

Resolves #326343

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by averyanalex** [comment]
I added myself as maintainer in #326213, so CI fails until merge

**Review by totoroot — APPROVED** [review]
LGTM

**Comment by averyanalex** [comment]
> I added myself as maintainer in #326213, so CI fails until merge

I cherry-picked maintainers update commit, now it should build

File: pkgs/by-name/np/npkill/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/np/npkill/package.nix:12]:
```suggestion
    repo = "npkill";
```

File: pkgs/by-name/np/npkill/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/np/npkill/package.nix:22]:
```suggestion
    description = "Easily find and remove old and heavy node_modules folders";
```

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**
